sycl/include/CL/sycl/queue.hpp

```diff
@@ -122,6 +122,89 @@ class queue {
     });
   }
 
+  // single_task version with a kernel represented as a lambda.
+  template <typename KernelName = detail::auto_name, typename KernelType>
+  event single_task(KernelType KernelFunc) {
+    return submit([&](handler &cgh) {
+      cgh.template single_task<KernelName, KernelType>(KernelFunc);
+    });
+  }
+
+  template <typename KernelName = detail::auto_name, typename KernelType>
```

Comment:
Please, add doxygen comment describing what the function does and its arguments.
Please, apply to all functions.

Reply:
Are there any examples of what this should look like in other files?

Reply:
Here is an example: https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/context.hpp.

Reply:
Done.